internal/consensus/replay_file.go

```diff
@@ -32,12 +32,22 @@ const (
 // replay messages interactively or all at once
 
 // replay the wal file
-func RunReplayFile(cfg config.BaseConfig, csConfig *config.ConsensusConfig, console bool) {
-	consensusState := newConsensusStateForReplay(cfg, csConfig)
+func RunReplayFile(
+	logger log.Logger,
+	cfg config.BaseConfig,
+	csConfig *config.ConsensusConfig,
+	console bool,
+) error {
+	consensusState, err := newConsensusStateForReplay(cfg, logger, csConfig)
+	if err != nil {
+		return err
+	}
 
 	if err := consensusState.ReplayFile(csConfig.WalFile(), console); err != nil {
-		tmos.Exit(fmt.Sprintf("Error during consensus replay: %v", err))
+		return fmt.Errorf("Error during consensus replay: %w", err)
```

Comment:
tangential, optional:

```suggestion
		return fmt.Errorf"consensus replay: %w", err)
```

Proximal motivation: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

But also "error during" doesn't add anything anyway.